src/bun.js/api/bun/subprocess.zig

```diff
@@ -2279,13 +2298,25 @@ pub const Subprocess = struct {
     }
 
     pub fn handleIPCClose(this: *Subprocess) void {
-        this.ipc_data = null;
+        Output.scoped(.IPC, false)("Subprocess#handleIPCClose\n", .{});
         this.updateHasPendingActivity();
+
+        const this_jsvalue = this.this_jsvalue;
+        this_jsvalue.ensureStillAlive();
+        if (this.on_disconnect_callback.trySwap()) |callback| {
```

Comment:
Do we not need to emit an optional error, like if it was closed due to the process sending invalid data?